### PR TITLE
da_revocation: align the revocation set generation algorithm with spec changes

### PR DESCRIPTION
**Pull request by shubhamdp**
There has been some updates to the revocation set generation algorithm in https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10308.

Add some helper function to interact with certificates.

#### Tests
Locally tested
```
python3 generate-revocation-set.py --use-main-net-http --log-level debug
python3 generate-revocation-set.py --use-test-net-http --log-level debug
```

### CODE REVIEW

File: credentials/generate-revocation-set.py
Comment (arkq) [inline credentials/generate-revocation-set.py:94]: Since you are catching the exception, the `bytes` output should be optional here.

File: credentials/generate-revocation-set.py
Comment (tcarmelveilleux) [inline credentials/generate-revocation-set.py:496]: Let's generate 1 set for each, even if empty.

File: credentials/generate-revocation-set.py
Comment (tcarmelveilleux) [inline credentials/generate-revocation-set.py:133]: This is too wide of an exception. Please catch the exception that the verification fails with.

File: credentials/generate-revocation-set.py
Comment (tcarmelveilleux) [inline credentials/generate-revocation-set.py:94]: `-> `Optional[bytes]`

Requires `from typing import Optional`

File: credentials/generate-revocation-set.py
Comment (tcarmelveilleux) [inline credentials/generate-revocation-set.py:105]: Please log in all exceptions what happened to help assist debugging.

File: credentials/generate-revocation-set.py
Comment (tcarmelveilleux) [inline credentials/generate-revocation-set.py:101]: Same here, missing optional.

File: credentials/generate-revocation-set.py
Comment (shubhamdp) [reply credentials/generate-revocation-set.py:108]: Logging the exceptions.

File: credentials/generate-revocation-set.py
Comment (shubhamdp) [reply credentials/generate-revocation-set.py:136]: logged.

File: credentials/generate-revocation-set.py
Comment (shubhamdp) [reply credentials/generate-revocation-set.py]: It was generating before, I though its an overhead so added that check, removed.

File: credentials/generate-revocation-set.py
Comment (shubhamdp) [reply credentials/generate-revocation-set.py]: Thanks for pointing it out, I had totally missed it. PTAL!

File: credentials/generate-revocation-set.py
Comment (shubhamdp) [reply credentials/generate-revocation-set.py:175]: Yes, its getting the issuer of the given cert rather than the PAA.

**Review by bh3000 — APPROVED** [review]

**Comment by bh3000** [comment]
Hey, just want to check in here. I am working a separate PR that will use this as a base.  Do you have an ETA for when you'll be able to unblock and merge this? Thanks!

**Comment by shubhamdp** [comment]
@bh3000 This is ready, just waiting for approvals. I have rebased on top of master. @tcarmelveilleux @arkq  can you please help with the review.

**Review by tcarmelveilleux — APPROVED** [review]

**Review by arkq — APPROVED** [review]

**Pull request merged**
